hydromt/components/geoms.py

```diff
@@ -0,0 +1,171 @@
+"""Table component."""
+
+import os
+from glob import glob
+from os.path import dirname, isdir, join
+from pathlib import Path
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    Optional,
+    Union,
+    cast,
+)
+
+import geopandas as gpd
+from geopandas import GeoDataFrame, GeoSeries
+
+from hydromt.components.base import ModelComponent
+from hydromt.hydromt_step import hydromt_step
+from hydromt.metadata_resolver import ConventionResolver
+
+if TYPE_CHECKING:
+    from hydromt.models.model import Model
+
+_DEFAULT_GEOMS_FILENAME = "geoms/{name}.geojson"
+
+
+class GeomsComponent(ModelComponent):
+    """A component to manage geo-spatial geometries."""
+
+    def __init__(self, model: "Model", filename: str = _DEFAULT_GEOMS_FILENAME):
+        """Initialize a GeomComponent.
+
+        Parameters
+        ----------
+        model: Model
+            HydroMT model instance
+        filename: str
+            The path to use for reading and writing of component data by default.
+            by default DEFAULT_GEOMS_FILENAME
```

Comment:
I dont think ``DEFAULT_GEOMS_FILENAME`` will print in this case. Can you use the actual value of the variable "geoms/{name}.geojson" ?
```suggestion
            by default "geoms/{name}.geojson" ie one file per geodataframe in the data dictionnary.
```